### PR TITLE
Initial review of ClinVar submissions 

### PR DESCRIPTION
**Pull request by twhetzel**
Initial review of ClinVar submissions on https://www.ncbi.nlm.nih.gov/clinvar/submitters/505999/

Related to #7773

### CODE REVIEW

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+Congenital titinopathy
+DAGLA-related disorder
+DHX9-related disorder
+DNM1L-related movement disorder
+DOHH-related disorder
+DONSON-related Meier-Gorlin syndrome
+DROSHA-related neurodevelopmental disorder
+DSP-related arrhythmogenic cardiomyopathy
+Decreased activity of mitochondrial ATP synthase complex
+Diarrhea
```
Comment (nicolevasilevsky) [inline src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt:38]: Are these all the new terms that will be added to Mondo? @twhetzel did you get synonyms to see if any of these terms existed in Mondo already from other sources besides medgen? ClinGen has requested gene-related diseases like these, so some of these terms could _potentially_ exist in Mondo that came from them (I spot checked a few but didn't see any examples).

But diarrhea is a synonym for MONDO:0001673 diarrheal disease

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+CACNA1A-related disorder
+CACNA1C-related disorder
+CDH2-related disorder
+CDK19-related disorder
+CDKL5-related disorder
+CHRNA3-related disorder
+COPB2-related disorder
+CRELD1-related disorder
+CTNNA1-associated FEVR
+Cerebellar atrophy
```
Comment (nicolevasilevsky) [inline src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt:28]: this is in HPO, so should we also add it as a disease in Mondo?
http://purl.obolibrary.org/obo/HP_0001272

**Comment by twhetzel** [comment]
@nicolevasilevsky this work is related to https://github.com/monarch-initiative/mondo/issues/7773 to scope out how many of these entries are not in Mondo. I do not know yet what the bigger picture is for what to do with terms. There may be some in the list that are in Mondo, e.g. those that do not start with a gene symbol, but others may be and would need to be reviewed later. This is just to scope out this first step.

**Comment by nicolevasilevsky** [comment]
Ok. I am not sure how to review this PR - will you be okay without my approval?

**Comment by twhetzel** [comment]
Sure, that's fine. I assigned it to Nico and wasn't sure if I could remove automatically assigned people as "Reviewers"

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/input/MedGenIDMappings.txt
Comment (matentzn) [inline src/scripts/notebooks/clinvar-udn-submission-analysis/data/input/MedGenIDMappings.txt:1]: Very large file to be added to the Mondo repo, could be in tmp. Not sure we always need to check in all data dependcies of all notebooks?

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+KIF5B-related osteogenesis imperfecta syndrome
+PAPSS1-related disorder
+MYCBP2-related disorder
+TMEM161B-related lissencephaly
+DHX9-related disorder
+DAGLA-related disorder
+SPTSSA-related disorder|Spastic paraplegia 90A, autosomal dominant
+ZNF865-related disorder
+SREBF2-related disorder
+Neurodevelopmental delay|Seizure
```
Comment (matentzn) [inline src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv:16]: What does it mean if there are two values?

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+PPP5C-related disorder|not provided
+CHARGE association
+MIR145-related multisystemic smooth muscle dysfunction
+ATP5F1A-related disorder
+NAV2-related neurodevelopmental condition
+Basal ganglia calcification, idiopathic, 5|not provided
+PHACTR1-related neurodevelopmental condition
+ATAD3 gene cluster related condition
+Spastic paraplegia 84, autosomal recessive|PI4KA-related disorder
+not provided|Primary ciliary dyskinesia
```
Comment (matentzn) [inline src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv:31]: What does "not provided" mean?

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt (hunk 2 above)
Comment (matentzn) [reply src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt:28]: The disease-phenotype issue will be significant for this work it seems!

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+TSPEAR-related disorder of tooth and hair follicle morphogenesis
+UNC45A-associated Cholestasis
+USP7-related disorder
+WHSC1-related disorder
+Wolf-Hirschhorn like syndrome
+ZBTB47-related disorder
+ZNF292-related neurodevelopmental condition
+ZNF331 deletion
+ZNF865-related disorder
+not specified
```
Comment (matentzn) [inline src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt:145]: Not a disease

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt
Comment (matentzn) [inline src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt:1]: Once you remove:

- 77 X-related disorder (where X is a gene)
- 35 X-related Y (where X is a gene, and Y is a disease concept)
- 8 X-associated Y (where X is a gene, and Y is a condition)
- 5 X-associated condition (where X is a gene)

you end up with these 18 classes that need to be manually curated:

```
1p13.3 deletion syndrome
ATAD3 gene cluster related condition
Adult hypophosphatasia
Bone mineral density quantitative trait locus 18
Cerebellar atrophy
Congenital titinopathy
Decreased activity of mitochondrial ATP synthase complex
Diarrhea
GP130-deficient hyper-IgE syndrome
Glutamine related condition
Impaired Hearing
Increased susceptibility to fractures
Motor neuropathy
NEMO deleted exon 5-autoinflammatory syndrome (NEMO-NDAS)
Predisposition to dissection
TRIP12 associated autism with facial dysmorphology
Wolf-Hirschhorn like syndrome
ZNF331 deletion
```

The first four groups need patterns and can be added without manual curation - but we first need to ensure if they are in scope for Mondo (whether the pattern is in scope).

**Review by matentzn — APPROVED** [review]
> ClinVar does not use Mondo labels for the disease name, e.g. Ataxia-telangiectasia syndrome is actually Ataxia telangiectasia syndrome.

You could do some preprocessing on the clinvar names prior to using oak search as a workaround.

The notebook looks great, i cant see any major issues blazing through!

Chris would probably say to use lexmatch on the diseases you identified as "not in Mondo" - I think there are some false positives. But in any case, these would need to become new term request issues, and the team needs to decide if they are in scope. Is `X-related disease` already a pattern in Mondo? 

Nice analysis!

**Comment by nicolevasilevsky** [comment]
@matentzn @twhetzel there is a pattern for x-related disease:
https://github.com/monarch-initiative/mondo/blob/master/src/patterns/dosdp-patterns/disease_series_by_gene.yaml

We definitely need to add logical definitions to these because a lot of terms will autoclassify as children of these terms

**Comment by twhetzel** [comment]
This first step was basically to filter out the known positives from the list (albeit knowing there would be some false positives) to scope out the curation effort. 

More discussion on the curation side will be needed to sort out an exact plan and the timing of this work is TBD.

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt (hunk 1 above)
Comment (twhetzel) [reply src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt:38]: Yeah, there will be some false positives in here. My aim was to filter out the known/easily found positives knowing the rest would need curation and we are familiar with our Mondo class naming policy and DOSDP patterns so can sort out this more easily. 

The choice of file I eventually used is in this [comment](https://github.com/monarch-initiative/mondo/issues/7773#issuecomment-2161452894). But again, this was intended as a quick check to filter out the easy things to help scope out the curation effort.

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt (hunk 5 above)
Comment (twhetzel) [reply src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_udn_submissions-no_mondo-disease_name_only.txt:145]: yeah, just left final file as-is

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv (hunk 3 above)
Comment (twhetzel) [reply src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv:16]: I'm not entirely sure and left for further curator review if/when this moves forward.

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv (hunk 4 above)
Comment (twhetzel) [reply src/scripts/notebooks/clinvar-udn-submission-analysis/data/output/clinvar_no_mondo.tsv:31]: not sure, can be addressed if/when this work moves forward

File: src/scripts/notebooks/clinvar-udn-submission-analysis/data/input/MedGenIDMappings.txt
Comment (twhetzel) [reply src/scripts/notebooks/clinvar-udn-submission-analysis/data/input/MedGenIDMappings.txt:1]: Good point, removed file and added line to fetch in notebook to gitignored tmp directory

**Pull request merged**